test/torchaudio_unittest/lfilter_autograd_test.py

```diff
@@ -0,0 +1,52 @@
+import torch
+import torchaudio.functional as F
+from parameterized import parameterized
+from torch.autograd import gradcheck
+from torchaudio_unittest import common_utils
+
+
+class AutogradLfilter(common_utils.TestBaseMixin):
```

Comment:
Thanks for adding auto-grad test. We were also considering adding the same kind of test (for native complex number support and verifying autograd support of existing functionals/transforms).

This class will be the model implementation of all the following autograd tests so I have couple of suggestions.

1. Please remove the helper methods `_get_a`, `_get_b` and `_get_waveform`, and add the equivalent in each function. (this appears to go against of principle of software engineering, but we want to keep unit tests independent each other as much as possible, and having this kind of helper function typically promotes other contributors to use the same and at the end it becomes difficult reason about the decision of parameters)
2. Please rename this class into `Autograd`, and have function name in each method. (like `test_lfilter_x_grad`)
3. Please split this into three different files like the other tests. (`impl`, `cpu_test`, `cuda_test`) This is for facebook's internal testing framework in which we enable/disable tests per file and we would like to split CPU and CUDA.

Reply:
Thanks for suggestions!